icu4c/source/common/cmemory.h

```diff
@@ -297,12 +297,16 @@ class MaybeStackArray {
      * Automatically allocates the heap array if the argument is larger than the stack capacity.
      * Intended for use when an approximate capacity is known at compile time but the true
      * capacity is not known until runtime.
-     *
-     * WARNING: does not report errors upon memory allocation failure, after
-     * which capacity will be stackCapacity, not the requested newCapacity.
      */
-    MaybeStackArray(int32_t newCapacity) : MaybeStackArray() {
-        if (capacity < newCapacity) { resize(newCapacity); }
+    MaybeStackArray(int32_t newCapacity, UErrorCode status) : MaybeStackArray() {
```

Comment:
This `: MaybeStackArray()` is new syntax to me. Looks like it invokes the default constructor before doing its specific work? That would be nice... otherwise this constructor would leave its fields undefined if the status is already an error.

Reply:
Self-education: https://en.cppreference.com/w/cpp/language/constructor#Delegating_constructor -- since C++11 -- very nice.